src/compositions/useVisibilityObserver.ts

```diff
@@ -25,6 +25,12 @@ export function useVisibilityObserver(element: Ref<Element | undefined>): UseVis
     }
   }, { immediate: true })
 
+  onUnmounted(() => {
+    if (element.value) {
+      observer.unobserve(element.value)
+    }
+  })
+
```

Comment:
unrelated fix to make sure elements are no longer observed after they are unmounted.

Reply:
Actually I'm going to make a separate PR for this so it will show up as a fix in the change log.